### DISCUSSION TITLE
Trouble in Fan Land

### DISCUSSION THREAD

**Discussion by helo-head** [HomeSpan/HomeSpan#135] [Q&A]
First let me say I was amazed at how easy it was to develop and deploy a ESP32 solution using HomeSpan. HomeSpan rocks, keep up the great work. 

I do have an issue that maybe someone can provide some guidance. It appears that  Characteristic::CurrentFanState (0=Off, 1=Idle, 2=On) has no graphical component in the Home application and as such Im not sure what it's used for. Currently the only way to turn on/off a fan is via the Characteristic::Active. There also appears to be no relationship between CurrentFanState and any other characteristic in that changing one has no impact on the other. 

I built an automated ventilation system and would actually like to leverage CurrentFanState. My thought was to use the characteristic Active to turn on/off the system and then use CurrentFanState to indicate the state of the ventilation system which supports all three States (off/idle/on) however as it stands today it does not seem like an option. 

Anyone thoughts on what Characteristic::CurrentFanState is useful for would be appreciated.

**Comment by HomeSpan** [comment]
The HAP-R2 documentation for things like this is unfortunately almost non-existent.  I experimented a bunch with CurrentFanState and  TargetFanState but to no avail.  The TargetFanState does create a method to switch from manual to auto (you can also see this if you click the settings icon for the fan in the Home App) but this does does not seem to have any effect.  I would have expected that if the fan is turned on (Active=true) but TargetFanState is set to "Auto" then the Home App would display the value from CurrentFanState, but it seems to ignore it.  You can see from typing 'd' in the Serial Monitor that the values are correctly sent.

The only other mention of CurrentFanState is in the Air Purifier, but according to the documentation it does not even appear the the Air Purifier uses it.

I've noticed that over time the Home App changes how it displays accessories (from iOS 12, to 13, to 14) but Apple never updates its HAP documentation to reflect changes that may alter the behavior of the some of the Characteristics.  For example, the Hold Characteristic for Window Shades seems like it was deprecated a while back even though it's still listed in the documentation.

**Reply by helo-head** [reply]
Thanks ... I decided I have enough granularity between  Activate and TargetFanState to do what I needed. Would be nice if all deprecated characteristics would be marked as such, but such is the state of life ...
